articles/a-beginners-guide-to-python/index.md

```diff
@@ -0,0 +1,207 @@
+
+In Python, we use the `def` keyword to declare a function. 
+An example of a python method is shown below.
+```
+def readData():
+    print('success')
+```
+The above function prints `success` when it’s invoked.
+We can also pass data to a method, perform some calculations, and return the results. This is demonstrated in the code snippet below.
+
+```
+def calculateTotal(chem, bio):
+    return chem+bio
+
+print(calculateTotal (90,80)
+```
+The `calculateTotal` method takes in two parameters (chem, bio). The function then returns the sum of the two values. It is important to take note of the data types when passing parameters. For instance, the `calculateTotal` method will not work when we pass in a string as a parameter. This is because the program cannot sum up an integer and a string. 
+As shown above, we can call the `calculateTotal` method directly from our print statement.
+```
+print(calculateTotal (90,80)
+```
+The `return` keyword ensures that the method returns a result after execution.
+
+Note that a function can also call another method. This is illustrated below.
+
+```python
+def readData(chem, bio):
+    return chem+bio
+
+def getTotal():
+    print(readData(90,80)) #calls the readData method
+
+getTotal()
+```
+
+### Understanding loops
+Loops are critical because they allow us to iterate through lists, check for different conditions, and continuously execute various statements. The main loops are `for` and `while`.
+
+1. For loops
+As noted, we can use a for loop to iterate through a list, as shown below.
+
+```
+Student_list=[“John Doore”,”Matu Smith”]
+for x in student_list:
+        print(x)
+```
+The above `for` loop will print every item in the student_list.
+
+2. While loops
+A while can help us check for a particular condition. For instance, while something is true specific statements can be executed. Here is an example of a while loop in action.
+
+```
+isChecked=false
+while isChecked == true:
+  print('Hallo there')
+```
+>Note that the above while loop will be executed indefinitely until isChecked is set to false. You can press ctrl+c to stop the loop.
+
+### Classes
+
+Classes are a vital component of object-oriented programming. When creating a class, you must use the `class` keyword. Other elements are then nested in the class. Here is an example of a python class.
+
+```python
+class Farmer: # a class with the name farmer
+    name="John" # A variable
+    produce ="1000kgs" # A variable
+
+farmer=Farmer() #instatiating the class as an object. 
+print(farmer.name) # accessing the properties of the Farmer class.
+```
+Classes can help as group things with similar characteristics. 
+
+We can also assign values to class variables using the `init` function.
+
+```python
+class Farmer:
+  def __init__(self, farmername, produce):
```

Comment:
Since you've discussed classes, please add a small explanation about how the `self` keyword works.